python/semantic_kernel/template_engine/protocols/prompt_templating_engine.py

```diff
@@ -1,18 +1,19 @@
 # Copyright (c) Microsoft. All rights reserved.
 
-from typing import TYPE_CHECKING, List, Optional, Protocol, runtime_checkable
+from abc import ABC
+from typing import TYPE_CHECKING, List, Optional
 
+from semantic_kernel.kernel_pydantic import KernelBaseModel
 from semantic_kernel.template_engine.blocks.block import Block
 
 if TYPE_CHECKING:
     from semantic_kernel import Kernel
     from semantic_kernel.functions.kernel_arguments import KernelArguments
 
 
-@runtime_checkable
-class PromptTemplatingEngine(Protocol):
+class PromptTemplatingEngine(KernelBaseModel, ABC):
```

Comment:
do we even need this whole class, in fact all protocols?

Reply:
I think we will have PromptTemplateEngines in the future when we support handlebars or the equivalent. Need to check how it's being used now.